samples: crypto: Fix mbedtls benchmark time calculations

rlubos opened the pull request:
mbedTLS benchmark used 32-bit arithmetics for time calculations
(nanosecond resolution), which could overflow on slower platforms
on more time-consuming benchmark tests. In result, the benchmark
could give incorrect timing information. Using 64-bit arithmetics
prevents this issue.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
samples: crypto: Fix mbedtls benchmark time calculations

mbedTLS benchmark used 32-bit arithmetics for time calculations
(nanosecond resolution), which could overflow on slower platforms
on more time-consuming benchmark tests. In result, the benchmark
could give incorrect timing information. Using 64-bit arithmetics
prevents this issue.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

aurel32 commented on samples/crypto/mbedtls_benchmark/src/benchmark.c:
```diff
@@ -179,7 +179,7 @@ do {                                                                  \
 	}                                                             \
 								      \
 	delta = k_cycle_get_32() - tsc;                               \
-	delta = SYS_CLOCK_HW_CYCLES_TO_NS(delta);                     \
```
If you really want to avoid overflow, I guess `k_cycle_get_32` should be changed into `k_cycle_get_64`.

rlubos replied:
I can't find this API, is it present in Zephyr headers?

aurel32 replied:
Oh right, I mixed that with `k_uptime_get` which has both a 32 and a 64-bit version. So looking at your change it seems the overflow doesn't happen at the cycle counter level, but in `SYS_CLOCK_HW_CYCLES_TO_NS`. Is it correct? If yes then your patch looks correct.

rlubos replied:
Correct, it's the nanosecond conversion that caused the overflow. All in all, with nanosecond resolution, `u32_t` overflows after ~4 seconds.

rveerama1 reviewed (approved):
LGTM.
@rlubos Can you add those boards to "samples/crypto/mbedtls_benchmark/sample.yaml" platform_whitelist.

rlubos committed:
samples: crypto: Whitelist nrf52840_pca10056 in mbedTLS benchmark tests

Add nrf52840_pca10056 board to the platform whitelist in mbedTLS
benchmark sample.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos commented:
@rveerama1 Done

rveerama1 commented:
Thank you.
